src/dfx/src/lib/metadata/dfx.rs

```diff
@@ -0,0 +1,18 @@
+//! A canister metadata with key "dfx"
+//!
+//! The cli tool dfx should consolidate its usage of canister metadata into this single section
+//! It's originally for pulling dependencies. But open to extend for other usage.
+
+use dfx_core::config::model::dfinity::PullReady;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+pub struct DfxMetadata {
+    pub pull_ready: PullReady,
```

Comment:
Since pull support isn't a requirement for all canisters, I suggest making this an option now.  Otherwise the first step anyone will have to do to add a field is to make this an Option.
This will also let us specify the error message ("canister {} is not ...") rather than starting from a serde error.
```suggestion
    #[serde(skip_serializing_if = "Option::is_none")]
    pub pull_ready: Option<PullReady>,
```